.github/workflows/release.yaml

```diff
@@ -100,6 +111,36 @@ jobs:
           AC_CERTIFICATE_PASSWORD: ${{ secrets.AC_CERTIFICATE_PASSWORD }}
           AC_APIKEY_P8_BASE64: ${{ secrets.AC_APIKEY_P8_BASE64 }}
 
+      - name: Create release tag and release notes
+        run: |
+          ref=HEAD
+          old_version="$(git describe --abbrev=0 "$ref^1")"
+
+          if [[ $DRY_RUN == true ]]; then
+            # Allow dry-run of branches to pass.
+            export CODER_IGNORE_MISSING_COMMIT_METADATA=1
+          fi
+
+          # Cache commit metadata.
+          . ./scripts/release/check_commit_metadata.sh "$old_version" "$ref"
+
+          # Create new release tag.
+          version="$(
+            ./scripts/release/increment_version_tag.sh \
+              ${{ (github.event.inputs.dry_run || github.event.inputs.snapshot) && '--dry-run' }} \
+              --ref "$ref" \
+              --${{ github.event.inputs.increment }}
+          )"
+
+          # Generate notes.
+          release_notes="$(./scripts/release/generate_release_notes.sh --old-version "$old_version" --new-version "$version" --ref "$ref")"
+          echo 'CODER_RELEASE_NOTES<<RN_EOF' >> $GITHUB_ENV
+          echo "$release_notes" >> $GITHUB_ENV
+          echo 'RN_EOF' >> $GITHUB_ENV
```

Comment:
This feels janky, what about writing it to a file and just using that file instead?

Reply:
This is official GH documentation for setting env variables that can be used between actions. It replaces the old `::set-` stuff.

Reply:
I'm aware, but storing the entire release notes in an environment variable seems like a misuse and a file should be used instead

Reply:
How-so? Environment variables and inputs are IMO the official way to pass information between steps, having a file maybe or maybe not being present between steps feels more magical to me 😅. As a case-in-point, Terraform stores a bunch of data in env variables.

Reply:
The file would always be present right? You could store the path to the file in an environment variable instead as well

Reply:
It's a theoretical use case, but for debugging purposes, we may want to run `env` and get the log output chopped if the notes are long.

Reply:
Switched to using a file now, still storing the file name in env, let me know if you think there's a more proper way.